server/s3_copy_object_action.cc

```diff
@@ -262,17 +113,24 @@ void S3CopyObjectAction::validate_copyobject_request() {
       return;
     }
   }
-  get_source_bucket_and_object();
-
-  if (source_bucket_name.empty() || source_object_name.empty()) {
+  if (additional_bucket_name.empty() || additional_object_name.empty()) {
     set_s3_error("InvalidArgument");
     send_response_to_s3_client();
+    return;
   } else if (if_source_and_destination_same()) {
     s3_put_action_state = S3PutObjectActionState::validationFailed;
     set_s3_error("InvalidRequest");
     send_response_to_s3_client();
+    return;
+  }
+  if (MaxCopyObjectSourceSize <
+      additional_object_metadata->get_content_length()) {
+    s3_put_action_state = S3PutObjectActionState::validationFailed;
+    set_s3_error("InvalidRequest");
+    send_response_to_s3_client();
```

Comment:
do we send specific error message here

Reply:
Yes

Reply:
The specified copy source is larger than the maximum allowable size for a copy source: 5368709120